addons/pinniped/post-deploy/pkg/configure/configure.go

```diff
@@ -178,11 +179,83 @@ func TKGAuthentication(c Clients) error {
 	ctx := context.Background()
 
 	inspector := inspect.Inspector{K8sClientset: c.K8SClientset, Context: ctx}
-	var tkgMetadata *inspect.TKGMetadata
-	if tkgMetadata, err = inspector.GetTKGMetadata(); err != nil {
-		zap.S().Error(err)
-		return err
+
+	// In classy clusters, the tkg-metadata CM is not currently created. We only use
+	// this CM for 1) cluster name and 2) cluster type. As a hack, let's skip reading
+	// this CM if we configure the cluster audience directly via a command line flag,
+	// because the flag tells us 1) what our audience is (so we don't have to use the
+	// cluster name) and 2) the cluster type (workload, since we don't pass this flag
+	// on the management cluster).
+	tkgMetadata := &inspect.TKGMetadata{}
+	tkgMetadata.Cluster = struct {
```

Comment:
nit: seems like some duplication with the anonymous struct defined in https://github.com/benjaminapetersen/tanzu-framework/blob/76677278958f4e337b49e9a7a1d64c159a970995/addons/pinniped/post-deploy/pkg/inspect/inspect.go#L52 in case you wanted to extract it and name it?

Reply:
+1 I definitely dislike this initialization & prob should have extracted immediately rather than going the dirty route.

Reply:
cleaned up!